pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/domain.go

```diff
@@ -62,3 +63,8 @@ func GenerateDomainName(name string, obj metav1.ObjectMeta, ingressConfig *v1bet
 
 	return buf.String(), nil
 }
+
+func IsValidHostURL(hostURL string) bool {
+	_, err := url.ParseRequestURI(hostURL)
```

Comment:
there is a package called `validation` in the apimachinery tool that could help with it.

IIRC: 
```go
validation.IsDNS1123Subdomain()
```

That might be useful here, as we are validating domains for ingress controller.

wdyt @israel-hdez @yuzisun ?

Reply:
I change the way for us to validate the domain or subdomain.
The tricky part is how to get the subdomain. The safest way is to get the domain name for the existing hosts that knative serving generated, remove it and append the new additional ingress domain to get the new hosts.